pkg/sqlcache/informer/listoption_indexer.go

```diff
@@ -248,11 +248,18 @@ func (l *ListOptionIndexer) deleteLabels(key string, tx transaction.Client) erro
 //   - a continue token, if there are more pages after the returned one
 //   - an error instead of all of the above if anything went wrong
 func (l *ListOptionIndexer) ListByOptions(ctx context.Context, lo ListOptions, partitions []partition.Partition, namespace string) (*unstructured.UnstructuredList, int, string, error) {
-	queryInfo, err := l.constructQuery(lo, partitions, namespace, db.Sanitize(l.GetName()))
+	dbName := db.Sanitize(l.GetName())
+	queryInfo, err := l.constructQuery(lo, partitions, namespace, dbName)
 	if err != nil {
 		return nil, 0, "", err
 	}
-	return l.executeQuery(ctx, queryInfo)
+	list, total, token, err := l.executeQuery(ctx, queryInfo)
+	if err != nil {
+		if isSpecialCaseQuery(lo, namespace, dbName) && removeSpecialCaseSituation(&lo, namespace, dbName) {
+			return l.ListByOptions(ctx, lo, partitions, namespace)
+		}
+	}
+	return list, total, token, err
```

Comment:
That is effectively ignoring the error from the query if the query is a "special" query. We should either detect the case that we want to ignore (not all errors) OR we should pre-emptively detect that we do / do not have the Project table.

Reply:
Yes, it does ignore the error. I wanted to see if the projects table exists, but I discovered through testing that sqlite isn't that smart -- if you try to do something like `select ... when <table "projects.cattle.io" exists> AND <some query about a field in "projects.cattle.io" …` you get an error message that table `projects.cattle.io` doesn't exist because although sqlite uses short-circuit AND-evaluation it still tries to compile references to fields in tables, and wants to see those fields up front.

I might be able to re-organize the code to make a separate query in the context, and will look into that. It actually wasn't obvious how to do that, based on where we create the ctx and db proxy objects.